### PR TITLE
Implemented generations, transactions, microBlocks store

### LINKED ISSUE #516: setup action from API call on SSR
child of #509

### LINKED ISSUE #517: setup store for generations
child of #509

### LINKED ISSUE #518: setup store for transactions
child of #509

### LINKED ISSUE #519: setup store for microblocks
child of #509

### PR DESCRIPTION
**Pull request by ror-shubham**
Closes #517, Closes #518, Closes #519, Closes #516

### CODE REVIEW

File: store/microBlock.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import isEqual from 'lodash/isEqual'
```
Comment (shekhar-shubhendu) [inline store/microBlock.js]: replace lodash with es6 counterpart.

File: store/microBlock.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    state.microBlock = microBlock
+  }
+}
+
+export const actions = wrapActionsWithResolvedNode({
+  async getMicroBlockFromHash ({ state, rootGetters: { node }, commit }, hash) {
+    const block = await node.api.getMicroBlockHeaderByHash(hash)
+    block.transactions = (await node.api.getMicroBlockTransactionsByHash(hash)).transactions
+
+    if (isEqual(state.block, block)) {
```
Comment (shekhar-shubhendu) [inline store/microBlock.js]: why is this check required?

File: store/transactions.js [hunk 3]
```diff
@@ -1,41 +1,37 @@
...
-    const transaction = await node.tx(hash, true)
-    commit('setTransaction', transaction)
-
-    return transaction
+export const actions = {
+  getLatestTransactions: async function ({ state, rootState: { nodeUrl, height }, commit, dispatch }, payload) {
+    const page = payload.page
+    const maxTransactions = payload.numTransactions
+    try {
+      const transactions = await axios.get(nodeUrl + 'middleware/transactions/interval/1/' + height.toString() + '?limit=' + maxTransactions + '&page=' + page)
```
Comment (shekhar-shubhendu) [inline store/transactions.js]: why start generation is `1`?

Comment (ror-shubham) [reply store/transactions.js]: I need the last `x` transactions. Currently middleware have an endpoint to get transactions from generation `a` to generation `b`. If I make a request from gen `1` to gen `height`, and put a limit of `x`, I get the latest `x` transactions.

File: store/microBlock.js (hunk 2 above)
Comment (ror-shubham) [reply store/microBlock.js]: It just prevents an extra commit to state. Now I think it's better if its not there.

File: store/microBlock.js (hunk 1 above)
Comment (ror-shubham) [reply store/microBlock.js]: removed it altogether

File: store/transactions.js (hunk 3 above)
Comment (shekhar-shubhendu) [reply store/transactions.js]: @ror-shubham. ok for now then. but not the best approach to take. maybe we can create an endpoint for this in the middleware.

**Comment by shekhar-shubhendu** [comment]
@ror-shubham  approved. nice work.

**Pull request merged**